crates/sui-kvstore/src/bigtable/client.rs

```diff
@@ -462,52 +488,116 @@ impl BigTableClient {
+            return result;
+        };
+
         let labels = [&self.client_name, table_name];
-        match &self.metrics {
-            None => result,
-            Some(metrics) => match result {
-                Err(e) => {
-                    metrics.kv_get_errors.with_label_values(&labels).inc();
-                    Err(e)
-                }
-                Ok(result) => {
-                    metrics
-                        .kv_get_batch_size
-                        .with_label_values(&labels)
-                        .observe(num_keys_requested as f64);
-                    if num_keys_requested > result.len() {
-                        metrics
-                            .kv_get_not_found
-                            .with_label_values(&labels)
-                            .inc_by((num_keys_requested - result.len()) as u64);
-                    }
-                    metrics
-                        .kv_get_success
-                        .with_label_values(&labels)
-                        .inc_by(result.len() as u64);
+        let Ok(rows) = &result else {
+            metrics.kv_get_errors.with_label_values(&labels).inc();
+            return result;
+        };
 
-                    metrics
-                        .kv_get_latency_ms
-                        .with_label_values(&labels)
-                        .observe(elapsed_ms);
-                    if num_keys_requested > 0 {
-                        metrics
-                            .kv_get_latency_ms_per_key
-                            .with_label_values(&labels)
-                            .observe(elapsed_ms / num_keys_requested as f64);
-                    }
-                    Ok(result)
-                }
-            },
+        metrics
+            .kv_get_batch_size
+            .with_label_values(&labels)
+            .observe(num_keys_requested as f64);
+
+        if num_keys_requested > rows.len() {
+            metrics
+                .kv_get_not_found
+                .with_label_values(&labels)
+                .inc_by((num_keys_requested - rows.len()) as u64);
+        }
+        metrics
+            .kv_get_success
+            .with_label_values(&labels)
+            .inc_by(rows.len() as u64);
+
+        metrics
+            .kv_get_latency_ms
+            .with_label_values(&labels)
+            .observe(elapsed_ms);
+
+        if num_keys_requested > 0 {
+            metrics
+                .kv_get_latency_ms_per_key
+                .with_label_values(&labels)
+                .observe(elapsed_ms / num_keys_requested as f64);
         }
+
+        result
+    }
+
+    pub async fn multi_get_column(
+        &mut self,
+        table_name: &str,
+        column_qualifier: &str,
+        keys: Vec<Vec<u8>>,
+    ) -> Result<Vec<Vec<(Bytes, Bytes)>>> {
+        let filter = RowFilter {
+            filter: Some(Filter::ColumnQualifierRegexFilter(
+                format!("^{column_qualifier}$").into_bytes(),
```

Comment:
nit: `Filter::ColumnQualifierRegexFilter(column_qualifier.into())` also works

Reply:
Is that the same thing? I'm introducing the beginning and end string anchors here. If you just call `.into()`, then it seems like it would be a substring match?

Reply:
Ah okay, great! If we don't need the anchors then agreed your suggestion is simpler!